Feat/prisma

labyrinth30 opened the pull request:
## 어떤 기능인가요?

> 추가하려는 기능에 대해 간결하게 설명해주세요

Prisma 세팅했습니다.

## 작업 상세 내용

- [x] 간단한 DB 설계(필요시 다른 테이블 추가 가능)
- [x] 프리즈마 로깅 추가
- [x] 도커 컴포즈 추가

## 참고할만한 자료(선택)
[prisma 공식 문서 - 로깅파트](https://www.prisma.io/docs/orm/prisma-client/observability-and-logging/logging)
package.json 속 prisma path 설정 같은 건 병호형 작업한 레포 보고 배워서 세팅했습니다.

labyrinth30 committed:
Feat: Install prisma

- Set Default prisma scheme
- Change prisma path

labyrinth30 committed:
Feat: Set Database With Docker Compose

- Create .example.env
- Create docker-compose.yml

labyrinth30 committed:
Feat: Add prisma setup

- Create prisma.service.ts
- Create yarn migrate:dev
- Add User Table (with migration files)

labyrinth30 committed:
Docs: Add setup instructions and commands

labyrinth30 committed:
Feat: Add logger on prisma

labyrinth30 committed:
Chore: Add newline at a end of file

labyrinth30 committed:
Feat: Add post table (with migration files)

Ho-s commented on .example.env:
```diff
@@ -0,0 +1,7 @@
+# DB
+DB_URL="postgresql://postgres:1q2w3e4r@localhost:5432/postgres?schema=public"
```
DB URL은 prisma schema 에쓰이는 것이고 나머지는 도커 env쪽에 쓰이는 것으로 보이는데
중복되는 사항이라, 중복 제거할 수 있는 방법이 있을까요?

labyrinth30 replied:
도커 컴포즈에 environment 속성으로 넣어볼까 생각하고 있습니다! 
```bash
environment:
  DATABASE_URL: "postgresql://${DB_USER}:${DB_PASSWORD}@${DB_HOST}:${DB_PORT}/${DB_NAME}?schema=${DB_SCHEMA}"
```
다만 이렇게 넣으니 이것도 조금 지저분한 것 같아 고민입니다.

Ho-s replied:
생긴건 지저분해도 직관적으로 이해는 되는 코드라 구조적으로는 지저분한 거 같지는 않아요.
중복을 없앤다는 측면에선 괜찮은 거 같은데 

이런 식으로 prisma랑 docker compose (postgresql) 맞춰볼까요?

labyrinth30 replied:
확인했습니다. 진행 후 커밋하겠습니다.

Ho-s reviewed (commented):
빠르고 정확한 작업!! 고생하셨습니다!!👍👍👍👍

hye-on commented on src/prisma/migrations/20250327042555_init/migration.sql:
```diff
@@ -0,0 +1,7 @@
+-- CreateTable
+CREATE TABLE "User" (
+    "id" SERIAL NOT NULL,
+    "name" TEXT,
+
+    CONSTRAINT "User_pkey" PRIMARY KEY ("id")
+);
```
`post`도 마이그레이션 파일에 추가해야될 것 같아! 스키마 파일에는 있어서!

hye-on replied:
수정 확인했습니다!

labyrinth30 committed:
Feat: Update User table structure and add Post table migration

labyrinth30 committed:
Merge branch 'main' into feat/prisma

labyrinth30 committed:
Feat: 환경변수 중복 제거

- docker-compose로 DB_URL 주입
- Dockerfile 생성
- .env 수정

labyrinth30 committed:
Chore: Build 문제 해결을 위한 라이브러리 설치

labyrinth30 committed:
Merge branch 'feat/prisma' of https://github.com/labyrinth30/mashup-nest-mercurius-fastify into feat/prisma

labyrinth30 committed:
Fix: Prisma Logger 설정 변경

- constructor에 넣은 logger를 바깥쪽으로 뺌

labyrinth30 committed:
Chore: DB_URL 환경변수 추가, PORT 수정

- 로컬에서 실행하기 위한 DB_URL 중복 변수 추가
- 기존 PORT 3000 -> 8000으로 변경

labyrinth30 committed:
Chore: Dockerfile 멀티 스테이지 빌드 사용하도록 업데이트

- 단일 스테이지에서 멀티 스테이지(빌더/러너)로 변경
- node:20-alpine으로 표준화
- @nestjs/cli 설치 단계 추가
- `npx prisma generate` 단계 추가
- 최종 이미지 아티팩트 최적화
- NODE_ENV=production 설정 및 노출 포트 8000으로 변경

labyrinth30 committed:
Chore: .dockerignore 추가

labyrinth30 committed:
Merge branch 'feat/prisma' of https://github.com/labyrinth30/mashup-nest-mercurius-fastify into feat/prisma

labyrinth30 committed:
Chore: @inquirer/core, cli-table3 라이브러리 제거, docker-compose volumes 디렉토리 수정

Ho-s reviewed (approved):
고생하셨습니다!! 추후 build 시에 prisma generate 넣어주는 등의 작업을 생각해보죠!

👍👍👍👍👍

hye-on reviewed (approved):
고생해쓰요!! 👍
